Source/IGListCollectionViewLayout.mm

```diff
@@ -0,0 +1,512 @@
+    NSInteger section = headerAttributes.indexPath.section;
+    NSInteger numberOfItemsInSection = [self.collectionView numberOfItemsInSection:section];
+    UICollectionViewLayoutAttributes *decorationViewAttributes = nil;
+
+    NSIndexPath *firstObjectIndexPath = [NSIndexPath indexPathForItem:0 inSection:section];
+    NSIndexPath *lastObjectIndexPath = [NSIndexPath indexPathForItem:MAX(0, (numberOfItemsInSection - 1)) inSection:section];
+
+    CGRect headerFrame = headerAttributes.frame;
+    CGRect firstObjectFrame;
+    CGRect lastObjectFrame;
+    if (numberOfItemsInSection > 0) {
+        firstObjectFrame = [self layoutAttributesForItemAtIndexPath:firstObjectIndexPath].frame;
+        lastObjectFrame = [self layoutAttributesForItemAtIndexPath:lastObjectIndexPath].frame;
+    } else {
+        firstObjectFrame = headerFrame;
+        lastObjectFrame = headerFrame;
+    }
+
+    decorationViewAttributes = [UICollectionViewLayoutAttributes layoutAttributesForDecorationViewOfKind:kind withIndexPath:firstObjectIndexPath];
+    decorationViewAttributes.frame = CGRectMake(CGRectGetMinX(headerFrame) - 1,
+                                                MIN(CGRectGetMinY(headerFrame), CGRectGetMinY(firstObjectFrame)) - 1,
+                                                CGRectGetWidth(headerFrame) + 2,
+                                                MAX(CGRectGetMaxY(lastObjectFrame), CGRectGetMaxY(headerFrame)) - MIN(CGRectGetMinY(headerFrame), CGRectGetMinY(firstObjectFrame)) + 2);
+    return decorationViewAttributes;
+}
+
+- (UICollectionViewLayoutAttributes *)layoutAttributesForDecorationViewOfKind:(NSString *)decorationViewKind atIndexPath:(NSIndexPath *)indexPath {
+    IGAssertMainThread();
+    IGParameterAssert(indexPath != nil);
+
+    UICollectionViewLayoutAttributes *attributes = _borderAttributesCache[indexPath];
+    if (attributes != nil) {
+        return attributes;
+    }
+    if (CGRectGetWidth(self.collectionView.bounds) < self.maximumContentWidth) {
+        return nil;
+    }
+    UICollectionViewLayoutAttributes *headerAttributes = [self layoutAttributesForSupplementaryViewOfKind:UICollectionElementKindSectionHeader atIndexPath:indexPath];
+    attributes = [self decorationBorderViewOfKind:decorationViewKind atIndexPath:indexPath withSectionHeaderAttributes:headerAttributes];
+    adjustZIndexForAttributes(attributes);
+    _borderAttributesCache[indexPath] = attributes;
+    return attributes;
+}
+
+- (CGSize)collectionViewContentSize {
+    IGAssertMainThread();
+
+    const NSInteger sectionCount = _sectionData.size();
+
+    if (sectionCount == 0) {
+        return CGSizeZero;
+    }
+
+    const IGSectionEntry section = _sectionData[sectionCount - 1];
+    const CGFloat height = CGRectGetMaxY(section.bounds) + section.insets.bottom;
+    return CGSizeMake(MIN(CGRectGetWidth(self.collectionView.bounds), self.maximumContentWidth), height);
```

Comment:
@rnystrom I found this [tutorial](https://www.raywenderlich.com/107439/uicollectionview-custom-layout-tutorial-pinterest) on Raywenderlich.
<img width="805" alt="screen shot 2017-02-04 at 11 10 35 pm" src="https://cloud.githubusercontent.com/assets/770482/22623624/507fb98c-eb2f-11e6-8c2b-884db532b5e4.png">
They took `contentInset` into account in this sample code. Idk if it is the common practice or not.